skglm/solvers/gram_cd.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from numba import njit
+from skglm.utils import AndersonAcceleration
+
+
+def gram_cd_solver(X, y, penalty, max_iter=20, use_acc=True, tol=1e-4, verbose=False):
+    """Run a Gram solver by reformulation the problem as below.
+
+    Minimize::
+        w.T @ Q @ w / (2*n_samples) - b.T @ w / n_samples + penalty(w)
+
+    where::
+        Q = X.T @ X
+        b = X.T @ y
+    """
+    n_samples, n_features = X.shape
+    XtX = X.T @ X / n_samples
+    Xty = X.T @ y / n_samples
+    all_features = np.arange(n_features)
+    p_objs_out = []
+
+    w = np.zeros(n_features)
+    XtXw = np.zeros(n_features)
+    # initial: grad = -Xty
+    opt = penalty.subdiff_distance(w, -Xty, all_features)
+    if use_acc:
+        accelerator = AndersonAcceleration(K=5)
+        w_acc = np.zeros(n_features)
+        XtXw_acc = np.zeros(n_features)
+
+    for t in range(max_iter):
+        # check convergences
+        stop_crit = np.max(opt)
+        if verbose:
+            p_obj = 0.5 * w @ XtXw - Xty @ w + penalty.value(w)
+            print(
+                f"Iteration {t+1}: {p_obj:.10f}, "
+                f"stopping crit: {stop_crit:.2e}"
+            )
+
+        if stop_crit <= tol:
+            if verbose:
+                print(f"Stopping criterion max violation: {stop_crit:.2e}")
+            break
+
+        # inplace update of w, XtXw, opt
+        _gram_cd_iter(XtX, Xty, w, XtXw, penalty, opt,
+                      all_features, n_updates=n_features)
+
+        # perform anderson extrapolation
+        if use_acc:
+            w_acc, XtXw_acc, is_extrapolated = accelerator.extrapolate(w, XtXw)
+
+            if is_extrapolated:
+                p_obj_acc = 0.5 * w_acc @ XtXw_acc - Xty @ w_acc + penalty.value(w_acc)
+                p_obj = 0.5 * w @ XtXw - Xty @ w + penalty.value(w)
+                if p_obj_acc < p_obj:
+                    w[:] = w_acc
+                    XtXw[:] = XtXw_acc
+
+        p_obj = 0.5 * w @ XtXw - Xty @ w + penalty.value(w)
+        p_objs_out.append(p_obj)
+    return w, p_objs_out, stop_crit
+
+
+@njit
+def _gram_cd_iter(XtX, Xty, w, XtXw, penalty, opt, ws, n_updates):
+    # inplace update of w, XtXw, opt
+    # perform greedy cd updates
+    for _ in range(n_updates):
+        grad = XtXw - Xty
```

Comment:
division by n_samples

OK, I just realised that XtX is X.T @ X / n_samples. That is clever to avoid divisions by n_samples but error prone (imagine someone adding just a few lines to this function in one year, they will likely think that XtX is X.T @ X). 

Can you modify the name ? XtX / n can be `gram` or `hess` for example